Build/QIR_validation.ps1

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = 'Stop'
+
+& "$PSScriptRoot/set-env.ps1"
+$all_ok = $True
+
+function Build-One {
+    param(
+        $project
+    );
+
+    $project_directory = $(Split-Path $project -Parent)
+    $project_name = $(Split-Path $project -LeafBase)
+
+    pushd $project_directory
+
+    Write-Host "##[info]Building $project..."
+    dotnet build $project `
+        -c $Env:BUILD_CONFIGURATION `
+        -v $Env:BUILD_VERBOSITY `
+        --no-dependencies `
+        /property:DefineConstants=$Env:ASSEMBLY_CONSTANTS `
+        /property:Version=$Env:ASSEMBLY_VERSION `
+        /property:QirGeneration="true"
+
+    if  ($LastExitCode -ne 0) {
+        Write-Host "##vso[task.logissue type=error;]Failed to build $project."
+        $script:all_ok = $False
+    } else {
+        qir-cli `
+            --dll (Join-Path bin $Env:BUILD_CONFIGURATION netcoreapp3.1 "${project_name}.dll") `
+            --exe qir
+        if  ($LastExitCode -ne 0) {
+            Write-Host "##vso[task.logissue type=error;]Failed to build $project."
+            $script:all_ok = $False
+        } else {
+            Get-ChildItem (Join-Path qir *__Interop.exe) `
```

Comment:
Can't you just pass the command line arguments here?

Reply:
I'm not sure how. I don't know if there is any way to automatically discover command line arguments for an given executable.

Reply:
done